core-plugins/src/main/java/co/cask/hydrator/plugin/batch/file/AbstractMetadataInputSplit.java

```diff
@@ -50,41 +56,53 @@ public AbstractMetadataInputSplit() {
   @Override
   public void write(DataOutput dataOutput) throws IOException {
     try {
-      // write obj summaries
+      // write number of files
       dataOutput.writeLong(this.getLength());
+
+      // write total number of file bytes
+      dataOutput.writeLong(this.getTotalBytes());
```

Comment:
this works, but most of the time you write the length of a collection and immediately follow it with the items in the collection. Let's do that here too and write the total bytes before we write the number of files.